lib/Doctrine/Deprecations/Deprecation.php

```diff
@@ -184,21 +156,37 @@ private static function delegateTriggerToBackend(string $message, array $backtra
 
             self::$logger->notice($message, $context);
         }
+
+        if (! ((self::$type & self::TYPE_TRIGGER_ERROR) > 0)) {
+            return;
+        }
+
+        $message .= sprintf(
+            ' (%s:%d called by %s:%d, %s, package %s)',
+            basename($backtrace[0]['file']),
```

Comment:
`basename()` is potentially slow and [locale-sensitive](https://bugs.php.net/62119), which means not able to handle all locales.

I'd suggest removing the file+line info from the message altogether, the userland error handler can compute them at will (many already do so.)

Reply:
PHP never ceases to amaze :) i will reimplement the function

I have the stacktrace anyways for outside detection and having this in the message gives a 80/20 useful information. 

For example PHPunit bridge requires additional configuration before it Shaws stacktraces.